### PR TITLE
Refine CI, use travis windows

### PR DESCRIPTION
**Pull request by Hoverbear**
Adopts using the Travis CI windows boxes for CI. Adopts a nearly identical CI to https://github.com/pingcap/raft-rs/pull/190.

AAppveyor failure is expected and shouldn't prevent merge. We can turn off appveyor after.

### CODE REVIEW

**Comment by kennytm** [comment]
Note that the [entire build](https://travis-ci.org/pingcap/fail-rs/builds/498447870) now takes 16 minutes. The Windows machine on Travis takes 10 minutes to run (compared with 1 minute on AppVeyor) because Travis uses rustup and we've got the bug rust-lang/rustup.rs#1540.

**Comment by Hoverbear** [comment]
I noticed this as well. :(

**Comment by brson** [comment]
I know @nrc was working on the upstream rustup bug at one point. Maybe this is something one of us should go fix. It's been one of the biggest rustup issues and it's not all that hard to fix _somehow_.

**Comment by nrc** [comment]
> I know @nrc was working on the upstream rustup bug at one point

I even have a PR, just needs some tests and some fixups. I guess this is motivation to get the damn thing finished

**Comment by BusyJay** [comment]
Aren't 9 CI checks too much for such a simple crate?

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
-      before_script:
-        - cargo fmt --all -- --check
-        - cargo clippy --all -- -D clippy::all
-    - rust: nightly
+install:
+  - if [[ $TRAVIS_RUST_VERSION == "stable" && $TRAVIS_OS_NAME == "linux" ]]; then rustup component add rustfmt; fi
+  - if [[ $TRAVIS_RUST_VERSION == "stable" && $TRAVIS_OS_NAME == "linux" ]]; then rustup component add clippy; fi
 
 script:
+<<<<<<< HEAD
```
Comment (BusyJay) [inline .travis.yml]: What's this?

Comment (Hoverbear) [reply .travis.yml]: A bad merge!

**Pull request merged**